src/Symfony/Component/DependencyInjection/Attribute/AutowireInline.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Loader\Configurator\InlineServiceConfigurator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Attribute to tell a parameter to inline a service with custom parameters.
+ *
+ * @author Ismail Özgün Turan <oezguen.turan@dadadev.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireInline extends Autowire
+{
+    /**
+     * @param class-string                                $class
+     * @param array                                       $args  Static parameters to pass to instance creation
+     * @param list<array{0: string, 1?: array, 2?: bool}> $calls A list of methods to be called after service initialization
+     * @param bool|class-string                           $lazy  Whether to use lazy-loading for this argument
+     */
+    public function __construct(
+        private readonly string $class,
+        private readonly null|string|array $factory = null,
+        private readonly array $args = [],
+        private readonly array $calls = [],
+        bool|string $lazy = false,
+    ) {
+        if (
+            \is_array($factory)
+            && !\is_string($factory[0])
+            && null !== $factory[0]
+            && !$factory[0] instanceof Reference
+        ) {
+            throw new LogicException(sprintf('#[%s] attribute defines a $factory which does not contain a callable.', static::class));
+        }
+
+        foreach ($this->calls as $call) {
+            if (!\is_array($call)) {
+                throw new LogicException(sprintf('#[%s] attribute defines a $calls entry which is not an array.', static::class));
+            }
+
+            if (!\array_key_exists(0, $call) || !\is_string($call[0])) {
+                throw new LogicException(sprintf('#[%s] attribute defines a $calls entry with invalid method name.', static::class));
+            }
+
+            if (\array_key_exists(1, $call) && !\is_array($call[1])) {
+                throw new LogicException(sprintf('#[%s] attribute defines a $calls entry with invalid arguments.', static::class));
+            }
+        }
+
+        parent::__construct($this->normalizeFactory($factory) ?? $class, lazy: $lazy);
+    }
+
+    public function buildDefinition(mixed $value, ?string $type, \ReflectionParameter $parameter): Definition
+    {
+        $definition = new Definition($this->class);
+        $configurator = (new InlineServiceConfigurator($definition))
```

Comment:
we don't want to use the configurator, let's play with $definition directly instead in this method

Reply:
I've removed the configurator from the `AutowireInline` attribute.